### PR TITLE
[RISC-V] Fix Reg Order for Float Registers

### PR DESCRIPTION
**Pull request by clamp03**
Order for some float registers are not set in `REG_VAR_ORDER_FLT`.
In release build, `regOrder` has invalid values and register allocator sets float registers differently every time.
Fix coreclr test `readytorun.coreroot_determinism.coreroot_determinism.coreroot_determinism` for risc-v release.

Part of https://github.com/dotnet/runtime/issues/84834
cc @dotnet/samsung

### CODE REVIEW

File: src/coreclr/jit/targetriscv64.h [hunk 1]
```diff
@@ -84,11 +84,11 @@
...
-  #define REG_VAR_ORDER_FLT        REG_F12,REG_F13,REG_F14,REG_F15,REG_F16,REG_F17,REG_F18,REG_F19, \
-                                   REG_F2,REG_F3,REG_F4,REG_F5,REG_F6,REG_F7,REG_F8,REG_F9,REG_F10, \
-                                   REG_F20,REG_F21,REG_F22,REG_F23, \
-                                   REG_F24,REG_F25,REG_F26,REG_F27,REG_F28,REG_F29,REG_F30,REG_F31, \
-                                   REG_F1,REG_F0
+  #define REG_VAR_ORDER_FLT        REG_F4, REG_F5, REG_F6, REG_F7, REG_F28, REG_F29, REG_F30, REG_F31, \
+                                   REG_F12, REG_F13, REG_F14, REG_F15, REG_F16, REG_F17, \
+                                   REG_F0, REG_F1, REG_F2, REG_F3, \
+                                   REG_F22, REG_F23, REG_F24, REG_F25, REG_F26, REG_F27, REG_F18, REG_F19, REG_F20, REG_F21, REG_F8, REG_F9, \
+                                   REG_F11, REG_F10
```
Comment (tomeksowi) [inline src/coreclr/jit/targetriscv64.h:91]: Could you explain where does this register order come from? In particular:
* the order of temporaries, why ft0-3 are placed after ft4-7, ft8-11, and fa2-7?
* the order of callee-saved, why it's fs6-11, fs3-5, fs0-1?

BTW, small nit: maybe use ABI register names like in REG_VAR_ORDER?

**Review by tomeksowi — APPROVED** [review]
LGTM, question about order.

File: src/coreclr/jit/targetriscv64.h (hunk 1 above)
Comment (clamp03) [reply src/coreclr/jit/targetriscv64.h:91]: I tried to follow LOONGARCH register order according to their types.
Okay. I will update to ABI register names.

Comment (clamp03) [reply src/coreclr/jit/targetriscv64.h:91]: As far as I know, there is no correct single order for REG_VAR_ORDER and REG_VAR_FLT_ORDER. And I just guess that it tries to avoid frequently used registers in the beginning in ORDER. So registers are mixed according to their types and index in other archs.
If I am wrong. please let me know.

Comment (tomeksowi) [reply src/coreclr/jit/targetriscv64.h:91]: That's probably because ARM and ARM64 don't have ABI names for float registers and LA started as a copy from it. But OK, it's just naming, doesn't matter much.

Comment (tomeksowi) [reply src/coreclr/jit/targetriscv64.h:91]: > As far as I know, there is no correct single order for REG_VAR_ORDER and REG_VAR_FLT_ORDER. And I just guess that it tries to avoid frequently used registers in the beginning in ORDER. So registers are mixed according to their types and index in other archs.

Yes, there's no correct order. We can figure out a better order for RISC-V when we start optimizing. For a bugfix this will do.

**Review by sirntar — APPROVED** [review]
LGTM

**Review by Bajtazar — APPROVED** [review]

**Comment by clamp03** [comment]
@jakobbotsch @jkotas Could you review this PR? Thank you.

File: src/coreclr/jit/lsra.h [hunk 2]
```diff
@@ -486,6 +486,7 @@ class RegRecord : public Referenceable
         previousInterval = nullptr;
         regNum           = REG_NA;
         isCalleeSave     = false;
+        regOrder         = UCHAR_MAX;
```
Comment (jakobbotsch) [inline src/coreclr/jit/lsra.h:489]: Why this change?

**Comment by risc-vv** [comment]
<details><summary>8d85c243is being scheduled for building and testing</summary>

GIT: `8d85c2430f8dfa2ae675eb764ca059991cd58e61`
REPO: [dotnet/runtime](https://github.com/dotnet/runtime)
BRANCH: main</details>

**Comment by risc-vv** [comment]
<details><summary>8d85c243 is being scheduled for building and testing</summary>

GIT: `8d85c2430f8dfa2ae675eb764ca059991cd58e61`
REPO: [dotnet/runtime](https://github.com/dotnet/runtime)
BRANCH: main</details>

File: src/coreclr/jit/lsra.h (hunk 2 above)
Comment (clamp03) [reply src/coreclr/jit/lsra.h:489]: If registers are not included in `REG_VAR_ORDER_FLT` and `REG_VAR_ORDER`, `regOrder` value for the registers are not set and can have a garbage value. Then LSRA can allocate differently each time and the test will fail. This code sets to lowest order for registers not included `REG_VAR_ORDER_*`.
I thought it is better to have the lowest order value than garbage value for any mistakes (missing regs in any other future archs).
If you think it is must to set all regs in `REG_VAR_ORDER_*`, I will remove this. Thank you.

**Comment by risc-vv** [comment]
<details><summary>8d85c243 is being scheduled for building and testing</summary>

GIT: `8d85c2430f8dfa2ae675eb764ca059991cd58e61`
REPO: [dotnet/runtime](https://github.com/dotnet/runtime)
BRANCH: main</details>

File: src/coreclr/jit/lsra.h (hunk 2 above)
Comment (jakobbotsch) [reply src/coreclr/jit/lsra.h:489]: Ok, let's keep it.. I just noticed that it showed up as a (very minor) throughput regression.

**Review by jakobbotsch — APPROVED** [review]
